jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cwc/editor/context/file/util/FileContextExtractorTest.kt

```diff
@@ -0,0 +1,132 @@
+import software.aws.toolkits.jetbrains.services.cwc.utility.EdtUtility
+
+class FileContextExtractorTest {
+
+    // Constructor parameters
+    private val mockFqnWebviewAdapter: FqnWebviewAdapter = mockk<FqnWebviewAdapter>(relaxed = true)
+    private val mockProject: Project = mockk<Project>(relaxed = true)
+    private val mockLanguageExtractor: LanguageExtractor = mockk<LanguageExtractor>(relaxed = true)
+
+    private val mockEditor: Editor = mockk<Editor>(relaxed = true)
+
+    /*
+    private lateinit var fileEditorManager: FileEditorManager
+    private lateinit var psiDocumentManager: PsiDocumentManager
+     */
+
+    private lateinit var fileContextExtractor: FileContextExtractor
+
+    @Before
+    fun setUp() {
+        fileContextExtractor = FileContextExtractor(mockFqnWebviewAdapter, mockProject, mockLanguageExtractor)
+
+        // Editor
+        mockkStatic(FileEditorManager::class)
+        every { FileEditorManager.getInstance(any()).selectedTextEditor } returns mockEditor
+
+        // computeInEdt
+        mockkObject(EdtUtility)
+        every { EdtUtility.runInEdt(any()) } answers {
+            firstArg<() -> Unit>().invoke()
+        }
+        every { EdtUtility.runReadAction<String> (any()) } answers {
+            firstArg<() -> String>().invoke()
+        }
+    }
+
+    @After
+    fun tearDown() {
+        clearMocks(
+            mockFqnWebviewAdapter,
+            mockProject,
+            mockLanguageExtractor,
+        )
+    }
+
+    @Test
+    fun `extract returns null when editor is null`() {
+        // Override return null editor
+        every { FileEditorManager.getInstance(any()).selectedTextEditor } returns null
+
+        runBlocking {
+            val fileContextExtractor = FileContextExtractor(mockFqnWebviewAdapter, mockProject, mockLanguageExtractor)
+
+            val result = fileContextExtractor.extract()
+
+            assertNull(result)
+        }
+    }
+
+    @Test
+    fun `extract returns FileContext when editor is not null`() {
+        val testFileLanguage = "java"
+        every { mockLanguageExtractor.extractLanguageNameFromCurrentFile(any(), any()) } returns testFileLanguage
+
+        val testFileText = "public class Test {}"
+        every { mockEditor.document.text } returns testFileText
+
+        mockkStatic(PsiDocumentManager::class)
+        val mockPsiFile = mockk<PsiFile>(relaxed = true)
+        every { PsiDocumentManager.getInstance(any()).getPsiFile(any()) } returns mockPsiFile
+        val testFilePath = "/path/to/file"
+        every { mockPsiFile.virtualFile.path } returns testFilePath
+
+        mockkObject(MatchPolicyExtractor)
+        coEvery { MatchPolicyExtractor.extractMatchPolicyFromCurrentFile(any(), any(), any(), any()) } returns null
+
+        // Act
+        val result = runBlocking {
+            fileContextExtractor.extract()
+        }
+
+        // Assert
+        assertNotNull(result)
+        assertEquals(testFileLanguage, result?.fileLanguage)
```

Comment:
```suggestion
        assertThat(result?.fileLanguage).isEqualTo(testFileLanguage)
```
Replace instances of junit assertions